src/features/api-keys/lib.ts

```diff
@@ -335,9 +335,10 @@ export const augmentReqApiKeyPermissions = <
 	T extends permissions.PermissionReq,
 >(
 	req: T,
-	...extraPermissions: string[]
+	extraPermissions: string[],
+	deepCloneRequest = true,
 ): T => {
-	const augmentedReq = _.clone(req);
+	const augmentedReq = deepCloneRequest ? _.clone(req) : req;
```

Comment:
`_.clone` isn't a deep clone so the variable naming is confusing, this is the difference between mutating the original request (and happening to return it) vs returning a new augmented one. If you want to mutate the original request then just doing
```ts
req.apiKey?.permissions?.push(...extraPermissions);
```
would do the same thing without additional unnecessary work

Reply:
You have point about the naming, so I will rename the parameter to `mutateRequest = false` and add some comments on why we have it.
We still do need to deep clone the apiKey & its permissions though afaict, since otherwise doing the push directly would pollute pine's `apiKeyPermissions` cache.